mscore/preferences.cpp

```diff
@@ -1154,13 +1154,13 @@ void PreferenceDialog::resetShortcutClicked()
     
 void PreferenceDialog::saveShortcutListClicked()
       {
-      QString saveFileName = QFileDialog::getSaveFileName(this, tr("Save Shortcuts"), "", tr("MuseScore Shortcuts File") + " (*.xml)");
+      QString saveFileName = QFileDialog::getSaveFileName(this, tr("Save Shortcuts"), "", tr("MuseScore Shortcuts File") + " (*.xml)", 0, preferences.nativeDialogs ? QFileDialog::Options() : QFileDialog::DontUseNativeDialog);
```

Comment:
I underrstand (more or less) why we use native dialogs in Mac and Windows and don't on Linux, I don#t understand why the code for that is so complicated when if can be done as simple as here?
Compare with e.g. https://github.com/musescore/MuseScore/blob/master/mscore/file.cpp#L935-L1000